mtp_api/apps/security/management/commands/bulk_unmonitor.py

```diff
@@ -0,0 +1,45 @@
+from django.contrib.auth import get_user_model
+from django.core.management import BaseCommand, CommandError
+
+from security.models import SavedSearch, PrisonerProfile, BankAccount, DebitCardSenderDetails
+
+User = get_user_model()
+
+
+class Command(BaseCommand):
+    def add_arguments(self, parser):
+        super().add_arguments(parser)
+        parser.add_argument('user', help='User id or username')
+        # TODO: add an option to move monitoring to a different user? requires deduplication
+
+    def handle(self, user, **options):
+        user = find_user(user)
+        if not user:
+            raise CommandError(f'User "{user}" not found')
+        self.stdout.write(f'Removing items monitored by {user.username} ({user.email})')
+
+        self.delete(SavedSearch, 'saved searches', user)
+        self.delete(PrisonerProfile.monitoring_users.through, 'monitored prisoners', user)
+        self.delete(BankAccount.monitoring_users.through, 'monitored bank account senders', user)
+        self.delete(DebitCardSenderDetails.monitoring_users.through, 'monitored debit card senders', user)
+
+    def delete(self, model, description, user):
+        self.stdout.write(f'Deleting all {description}…')
+        count, deleted_models = model.objects.filter(user=user).delete()
+        if count:
+            for model, count in deleted_models.items():
+                self.stdout.write(f'Deleted {count} × {model}')
```

Comment:
Do we loop through the result of `delete()` because this could potentially delete other associated records and this will be shown here as well?

Maybe this is why this is not equivalent to the simpler?

```python
        count, _ = model.objects.filter(user=user).delete()
        if count:
             self.stdout.write(f'Deleted {count} × {model}')
```

Reply:
the delete cascades so ends up with several models being deleted (e.g. SavedSearch deletes SavedFilter as well) so there are several lines of output generated